agent/command/papertrail_trace.go

```diff
@@ -73,12 +68,61 @@ func (t *papertrailTrace) Execute(ctx context.Context,
 			return errors.Wrap(err, "running trace")
 		}
 
-		logger.Task().Infof("Successfully traced '%s' (sha256://%s)", fullname, sha256sum)
+		logger.Task().Infof("Successfully traced '%s' (sha256://%s)", f.filename, f.sha256)
 	}
 
 	return nil
 }
 
+type papertrailTraceFile struct {
+	filename string
+	sha256   string
+}
+
+func getTraceFiles(workdir string, patterns []string) ([]papertrailTraceFile, error) {
+	files := make([]papertrailTraceFile, 0, len(patterns))
+
+	seen := make(map[string]string)
+
+	for _, path := range patterns {
+		patternpath := filepath.Join(workdir, path)
+
+		matches, err := filepath.Glob(patternpath)
+		if err != nil {
+			return nil, errors.Wrap(err, "globbing filepath")
+		}
+
+		for _, matchpath := range matches {
+			sha256sum, err := getSHA256(matchpath)
+			if err != nil {
+				return nil, errors.Wrap(err, "getting sha256")
+			}
+
+			basename := filepath.Base(matchpath)
+
+			// santity check that the filename we're tracing is legitimate
+			if basename == "." || basename == string(filepath.Separator) {
```

Comment:
Just a suggestion, but it seems like it might be a little clearer, rather than checking these specific invalid file patterns, to instead open the file (which is already done in `getSHA256`) and just check if `f.IsDir()`.

Reply:
You're probably right that this check is unnecessary in the end because I already checked legitimacy when I read it in earlier, I'll just remove it.